lib/VM/JSLib/RegExp.cpp

```diff
@@ -460,6 +461,65 @@ static void createGroupsObject(
   JSObject::setNamedSlotValueUnsafe(matchObj.get(), runtime, groupsDesc, shv);
 }
 
+static void createIndicesArray(
+    Runtime &runtime,
+    Handle<JSArray> matchObj,
+    Handle<JSObject> mappingObj,
+    Handle<JSArray> indices) {
+  // If there are no capture groups, then set groups to undefined.
+  if (!mappingObj) {
+    auto executionStatus = JSObject::defineNewOwnProperty(
+        indices,
+        runtime,
+        Predefined::getSymbolID(Predefined::groups),
+        PropertyFlags::defaultNewNamedPropertyFlags(),
+        Runtime::getUndefinedValue());
+    assert(
```

Comment:
These `assert`s should not be written this way. You should check the result of the `defineProperty` and return in an error in a regular `if`. Also remember to wrap these in the appropriate `LLVM_(UN)LIKELY` For example,

```
if (LLVM_UNLIKELY(JSObject::defineNewOwnProperty(...) == ExecutionStatus::EXCEPTION)){
  return ExecutionStatus::EXCEPTION;
}
```

Reply:
I tried to do this, but since my `createIndicesArray` is a `void` method this approach wouldn't work, so I followed the same way was done to `createGroupsObject` which is also a `void` method and uses `assert`. What's your advice? Change the method's signature to return something like `CallResult<void>` ?

Reply:
You change the type to `ExecutionStatus`.